### PR TITLE
Claim airdrops module

### PR DESCRIPTION
**Pull request by godekina**
@djeck1432 kindly review the implementation of the claim_airdrops module from issue: https://github.com/djeck1432/spotnet/issues/137

### CODE REVIEW

File: web_app/tasks/claim_airdrops.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+from typing import List
+
+from web_app.contract_tools.airdrop import ZkLendAirdrop
+from web_app.contract_tools.blockchain_call import StarknetClient
+from web_app.db.crud import AirDropDBConnector
+
+logger = logging.getLogger(__name__)
+
+
+##1. fetch all airdops from AirDropDBConnector.get_all_unclaimed()
```
Comment (djeck1432) [inline web_app/tasks/claim_airdrops.py:17]: Remove CHATGPT comment please

File: web_app/tasks/claim_airdrops.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    def __init__(self):
+        """
+        Initializes the AirdropClaimer with database and Starknet client instances.
+        """
+        self.db_connector = AirDropDBConnector()
+        self.starknet_clinet = StarknetClient()
+        self.zk_lend_airdrop = ZkLendAirdrop()
+
+    ##2. create a function to claim airdrops using Starknetclient
```
Comment (djeck1432) [inline web_app/tasks/claim_airdrops.py:31]: Remove CHATGPT comment please

File: web_app/tasks/claim_airdrops.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                    user_contract_address, proof
+                )
+
+                if claim_succesful:
+                    self.db_connector.save_claim_data(airdrop.id, airdrop.amount)
+                    logger.info("Airdrop %s claimed succesfully.", airdrop.id)
+            except Exception as e:
+                logger.error("Error claiming airdrop %s: %s", airdrop.id, e)
+
+    async def _claim_airdrop(self, contract_address: str, proof: List[int]) -> bool:
```
Comment (djeck1432) [inline web_app/tasks/claim_airdrops.py:55]: `proof: List[str]` there are list of strings

File: web_app/tasks/claim_airdrops.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        """
+        Claims a single airdrop by making a contract call on the Starknet blockchain.
+        """
+        calldata = [100] + proof
```
Comment (djeck1432) [inline web_app/tasks/claim_airdrops.py:59]: what for this `[100]`?

File: web_app/tasks/claim_airdrops.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        try:
+            # Starknet contract call
+            await self.starknet_client._func_call(
+                addr=self.starknet_client._convert_address(contract_address),
+                selector="claim",
+                calldata=calldata,
+            )
+            return True
+        except Exception as e:
```
Comment (djeck1432) [inline web_app/tasks/claim_airdrops.py:68]: Which particular exception you are trying to catch, please, be more specific

**Review by djeck1432 — CHANGES_REQUESTED** [review]

File: web_app/tasks/claim_airdrops.py (hunk 4 above)
Comment (godekina) [reply web_app/tasks/claim_airdrops.py:55]: this is a placeholder for the calldata list, but will be removed.

File: web_app/tasks/claim_airdrops.py (hunk 5 above)
Comment (godekina) [reply web_app/tasks/claim_airdrops.py:63]: I will add the specific exceptions, may I add them all in a tuple?

Comment (djeck1432) [reply web_app/tasks/claim_airdrops.py:63]: Yeah, you can do that

File: web_app/tasks/claim_airdrops.py (hunk 4 above)
Comment (djeck1432) [reply web_app/tasks/claim_airdrops.py:55]: Please remove it

**Review by djeck1432 — CHANGES_REQUESTED** [review]

File: web_app/tasks/claim_airdrops.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+            except Timeout as te:
+                logger.error("Timeout during claim for airdrop %s: %s", airdrop.id, te)
+            except Exception as e:
+                logger.error("Unexpected error claiming airdrop %s: %s", airdrop.id, e)
+
+    async def _claim_airdrop(self, contract_address: str, proof: List[str]) -> bool:
+        """
+        Claims a single airdrop by making a contract call on the Starknet blockchain.
+        """
+        calldata = [] + proof
```
Comment (djeck1432) [inline web_app/tasks/claim_airdrops.py:73]: why do we need it, why just insert simple `proofs` what is actually is a `list`?

File: web_app/tasks/claim_airdrops.py [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+                    "Network connection error during claim for airdrop %s: %s",
+                    airdrop.id,
+                    ce,
+                )
+            except Timeout as te:
+                logger.error("Timeout during claim for airdrop %s: %s", airdrop.id, te)
+            except Exception as e:
+                logger.error("Unexpected error claiming airdrop %s: %s", airdrop.id, e)
+
+    async def _claim_airdrop(self, contract_address: str, proof: List[str]) -> bool:
```
Comment (djeck1432) [inline web_app/tasks/claim_airdrops.py:69]: `proofs` not `proof`

File: web_app/tasks/claim_airdrops.py [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+    async def claim_airdrops(self) -> None:
+        """
+        Retrieves unclaimed airdrops, attempts to claim them on the Starknet blockchain,
+        and updates the database if the claim is successful.
+        """
+        unclaimed_airdrops = self.db_connector.get_all_unclaimed()
+        for airdrop in unclaimed_airdrops:
+            try:
+                user_contract_address = airdrop.user.contract_address
+                proof = self.zk_lend_airdrop.get_contract_airdrop(user_contract_address)
```
Comment (djeck1432) [inline web_app/tasks/claim_airdrops.py:41]: `proofs`

**Review by djeck1432 — CHANGES_REQUESTED** [review]

**Comment by djeck1432** [comment]
@godekina Any updates?

**Review by djeck1432 — APPROVED** [review]
Looks good to me

**Pull request merged**